src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -145,30 +144,62 @@ public ObservableList<Student> getInternalList() {
         LocalDate currDate = LocalDate.now();
-        List<Student> list = internalList
+        List<Class> list = internalList
                 .stream()
                 .filter(student -> student.getAClass().startTime != null
                         && student.getAClass().endTime != null
                         && student.getAClass().date != null
-                        && student.getAClass().date.compareTo(currDate) >= 0
-                        && student.getAClass().startTime.compareTo(LocalTime.now()) >= 0)
-                .sorted(Student::compareToByClassStartTimeAsc)
+                        && student.getAClass().date.compareTo(currDate) >= 0)
+                .sorted(Student::compareToByClassAsc)
+                .map((element) -> element.getAClass())
                 .collect(Collectors.toList());
-        Class newClass = new Class();
+        Class newClass = null;
+
+
         if (list.size() == 0) {
-            newClass = new Class(currDate, tr.startTimeRange,
-                    tr.startTimeRange.plusMinutes(tr.duration));
+            // Case where there is only no classes
+            LocalTime startTime = tr.startTimeRange;
+            LocalTime endTime = startTime.plusMinutes(tr.duration);
+            // Link to design: https://arc.net/e/B15E0B60-817D-4B24-8397-FC0E0B37F8C1
+            if (currTime.compareTo(tr.endTimeRange) >= 0) {
+                // this is the situation where the current time is after and including the endTimeRange
+                newClass = new Class(currDate.plusDays(1), startTime, endTime);
+            } else if (currTime.compareTo(tr.startTimeRange) <= 0) {
+                // this is the situation where the current time is before the startTimeRange
+                newClass = new Class(currDate, startTime, endTime);
+            } else {
+                // this is the situation where the current time is after the startTimeRange and before the endTimeRange
+                LocalTime startTimeFromCurrTime = currTime;
+                LocalTime endTimeFromCurrTime = startTimeFromCurrTime.plusMinutes(tr.duration);
+
+                if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                    newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                } else {
+                    newClass = new Class(currDate.plusDays(1), startTime, endTime);
+                }
+            }
+
             return newClass;
         } else if (list.size() == 1) {
-            return findAvailableClassWithSingleRecord(tr, currDate, list);
+            // Link to design for 1 class: https://arc.net/e/4F0065DB-C858-4B6D-91D0-50B2879AC26F
+            return findAvailableClassWithSingleRecord(tr, currDate, list, currTime);
         }
 
+
+        // Link to design for more than 2 classes: https://arc.net/e/8D5B34DF-70C5-4A22-8D26-4BFC6CEEABC5
         for (int i = 0; i < list.size(); i++) {
```

Comment:
As I am not done reading the code, so cannot really tell which part goes wrong. But I have found the issue here:
current time: 0023
classes: [2022-11-07 0000-0100, 2022-11-07 0800-0900]
`avail 0000-2300 100` expects `2022-11-07 0100-0240` but receives `2022-11-07 0023-0203`

Reply:
Sure, I'll look into it. In the future, there is no need to try to find the bug for me. I can trace the code to try to solve the bug! Thank you so much for pointing it out in the past though!